src/lib/comms/sol-http-server-impl-microhttpd.c

```diff
@@ -1035,12 +1052,28 @@ sol_http_server_new(uint16_t port)
 
     server->buf_size = SOL_HTTP_REQUEST_BUFFER_SIZE;
 
-    server->daemon = MHD_start_daemon(MHD_USE_SUSPEND_RESUME,
-        port, NULL, NULL,
-        http_server_handler, server,
-        MHD_OPTION_NOTIFY_CONNECTION, notify_connection_cb, server,
-        MHD_OPTION_NOTIFY_COMPLETED, notify_connection_finished_cb, server,
-        MHD_OPTION_END);
+    if (config->security.cert && config->security.key) {
+        struct sol_blob *cert_contents, *key_contents;
+        cert_contents = sol_cert_get_contents(config->security.cert);
+        key_contents = sol_cert_get_contents(config->security.key);
```

Comment:
not necessary, microhttpd will fail in this case, as the error treatment will be the same ...